get_local_image_transform_command.py

```diff
@@ -0,0 +1,154 @@
+# you may not use this file except in compliance with the Licence.
+# You may obtain a copy of the Licence at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Licence is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Licence for the specific language governing permissions and
+# limitations under the Licence.
+
+import sys
+import os
+
+import requests
+import argparse
+
+"""
+Source and target spaces can be one of the 4 spaces:
+"MNI 152 ICBM 2009c Nonlinear Asymmetric"
+"MNI Colin 27"
+"Big Brain (Histology)"
+"Infant Atlas"
+"""
+SOURCE_SPACE_DEFAULT = "MNI 152 ICBM 2009c Nonlinear Asymmetric"
+TARGET_SPACE_DEFAULT = "Big Brain (Histology)"
+
+
+SERVER_ADDRESS_DEFAULT = "http://0.0.0.0:8080"
+# SERVER_ADDRESS_DEFAULT = "http://172.17.0.1:8080"
+# SERVER_ADDRESS_DEFAULT = "https://hbp-spatial-backend.apps-dev.hbp.eu"
+INSTANCE_PATH = "/instance"
+
+
+def is_transform(str_in):
+    """Determines if str_in is the name of a transformation file
+    """
+    return '.ima' in str_in \
+        or '.trm' in str_in
+
+
+def add_path_to_transform(str_in):
+    """Adds as prefix INSTANCE_PATH to str_in if str_in is a transformaton file
+    """
+    if is_transform(str_in):
+        return f'{INSTANCE_PATH}/{str_in}'
+    else:
+        return str_in
+
+
+def get_command(server_address, source_space, target_space):
+    """Gets the image transform commmand from the server
+    """
+    request_string = \
+        f'{server_address}/v1/'\
+        f'get-image-transform-command?source_space={source_space}'\
+        f'&target_space={target_space}&input_coords=auto'
+    return requests.get(request_string)
+
+
+def format_command(command_from_server, input_file, output_file, extra):
+    """Formats the command to a string that can be run on a bash shell
+    """
+    command_as_list = command_from_server.json()['transform_command']
+    command_as_list = [add_path_to_transform(x) for x in command_as_list]
+    command_as_list.extend(['-i', input_file])
+    command_as_list.extend(['-o', output_file])
+    command_as_list.extend(extra)
+    return ' '.join(command_as_list)
+
+
+def parse_args(argv):
+    """Function parsing command-line arguments
+
+    Args:
+        argv: a list containing command line arguments
+
+    Returns:
+        args: arguments used by this program
+        extra: extra arguments passed to AimsApplyTransform
+    """
+
+    # Parse command line arguments
+    parser = argparse.ArgumentParser(
+        prog='get_local_image_transform_command.py',
+        description='Gets command to transform image among templates')
+    parser.add_argument(
+        "-a", "--server_address", type=str, default=SERVER_ADDRESS_DEFAULT,
+        help='Server address from which to get the transform command. '
+             'From inside docker, it is http://0.0.0.0:8080 .'
```

Comment:
`0.0.0.0:8080` is not a valid IP address, rather, it is a convention for telling the web server to listen on all interfaces. If you want to connect from inside Docker, you should use `localhost:8080`, so I think the doc should be changed to this value.